### ISSUE TITLE
Bejeweled.swf content works but isn't quite right ...

### ISSUE THREAD

**Issue by SeaMac** [ruffle-rs/ruffle#3604]
File location is https://thetruthabouthemp.com/hempity/flash/Bejeweled.swf - Working Demo Page is 
https://thetruthabouthemp.com/arcade/Bejeweled.html 
[Bejeweled.swf.zip](https://github.com/ruffle-rs/ruffle/files/6119495/Bejeweled.swf.zip)
Testing with Ruffle Nightly 3-10-2021 on Firefox

Problem #2710 Text Labels on Buttons Missing

And a little problem with the game logic. When the game reaches "No More Moves" and "Game Over" there is one button to click (I presume it is supposed to read "Continue") - when a new Board is Spawned it has "No More Moves" left ... and the game is over in 3 seconds with a score of zero. Click the button again to watch it spawn the exact same fatal board. Again. And again. 

If i had to make a stab of a guess I'd say there was a problem generating a 'random seed' for the new board.

This should be a popular game once the labels return and new games spawn OK. I play with it regularly even while broken in my little arcade / Ruffle Testbed at https://thetruthabouthemp.com/arcade/ 

(Chuck Jones last cartoon project in 2001 "Thomas Timberwolf" was 13 cartoons written in SWF and I found them and hosted them and ruffle WORKS PERFECTLY on ALL of Them. Kudos!!!! https://thetruthabouthemp.com/arcade/wolf/index.html )

**Comment by n0samu** [comment]
The missing text issue was actually caused by a missing XML file on the original site. Here is an alternative link that includes the required XML file: http://www.match3games.com/uploaded/swf/Bejeweled.swf

The "no more moves" randomization problem is still occurring though:
https://user-images.githubusercontent.com/71368227/152672872-cd03d094-85b8-46d2-835d-64b7bf5e420e.mp4

Tested with latest (2022-02-06) extension.

**Comment by Toad06** [comment]
I think this is another instance of #3839, but not sure. Unfortunately the game doesn't load at all when compiling with #5492, so I can't verify.

**Comment by SeaMac** [comment]
At least the file CAN be Played with Ruffle 3-26-2021 as shown here: https://thetruthabouthemp.com/arcade/Bejeweled.html but a version of Ruffle very SOON after that date broke it forever after ... now it just loads a splash screen and nothing else. 
Since you HAD a version of Ruffle that WORKED I went back to using it. 

![image](https://user-images.githubusercontent.com/5051781/152700689-b64773a5-78c3-4900-9234-60ee94ec67ac.png)

But no version of Ruffle since then has worked for this file. The version of Ruffle, most compatible ... with the largest number of files I have, is from last March! Why did it load and play back THEN and not since? Seems like Ruffle peaked in March last year. I know it did Not and you guys will get this sorted out eventually. When a version of Ruffle more recent than 2021-03-26 can play this file I'll replace the version on my server with THAT Version. 

I wish I could help code this (I dream in flowcharts but don't 'speak' the "Rust" dialect) ... but I sure the Heck can Beta Test it for you. I've got an old Mac workstation with 10.13 on it ... I'll never have the latest greatest hardware or software ... but I know you need it tested on a variety of platforms and versions of OSes too. If I can test some file out: speak up ... I'll be glad to.

**Comment by Toad06** [comment]
I'm afraid to say that the issue is actually in the old version you are using. As @n0samu mentioned, the game loads an XML file (named `bejeweled.xml`). This file includes the localization strings and it _must_ be loaded successfully to let the game start.

Here is a part of the ActionScript code that triggers when the XML file is loaded:

```
   this.LocalizedXML.onLoad = function(success)
   {
      if(!success)
      {
         // Failed to load the file (eg. file not found), stops execution.
      }
      else
      {
         // Got the file. Parse it to get the strings, etc. and set the global
         // variable below to say that the game can start.
         _root.localizationFinished = true;
      }
   };
```
Previously Ruffle didn't check the status code of the files it fetched, so a status of 404 (file not found) was incorrectly considered as a success. This was fixed by #3800 and shipped in nightly-2021-04-07.

To make the game start again, [download this XML file](https://www.match3games.com/uploaded/swf/bejeweled.xml) and upload it to your server, in the same directory as your HTML file.

**Comment by SeaMac** [comment]
> I'm afraid to say that the issue is actually in the old version you are using. As @n0samu mentioned, the game loads an XML file (named `bejeweled.xml`). This file includes the localization strings and it _must_ be loaded successfully to let the game start.
> 
> Here is a part of the ActionScript code that triggers when the XML file is loaded:
> 
> ```
>    this.LocalizedXML.onLoad = function(success)
>    {
>       if(!success)
>       {
>          // Failed to load the file (eg. file not found), stops execution.
>       }
>       else
>       {
>          // Got the file. Parse it to get the strings, etc. and set the global
>          // variable below to say that the game can start.
>          _root.localizationFinished = true;
>       }
>    };
> ```
> 
> Previously Ruffle didn't check the status code of the files it fetched, so a status of 404 (file not found) was incorrectly considered as a success. This was fixed by #3800 and shipped in nightly-2021-04-07.
> 
> To make the game start again, [download this XML file](https://www.match3games.com/uploaded/swf/bejeweled.xml) and upload it to your server, in the same directory as your HTML file.

**YES!!!!**
And Thank you very much for explaining that to me in little bitty words ... I'm autistic. 
You gave Specific directions ... and we have a CURE!!

Our testbed is located at https://thetruthabouthemp.com/arcade/ 17 flash game files and a sub-directory of 13 flash animations (With "Mini Games" on each Cartoon!)

Bejeweled is working Perfectly EXCEPT for when you make your last match and the restart shuffle fails.

As an upshot to us all: we now have 17 flash game files and 13 flash Cartoons by Chuck Jones (Each with a Game built in) ... hosted with TONIGHT'S Build 2022-02-06 - all 31 of which I shall proceed to test out ... those latter 13 files being the LAST animation project of the legendary director "Chuck Jones" ... Called "Thomas Timberwolf"  
https://thetruthabouthemp.com/arcade/wolf/index.php 
(The "Timberwolf" Logo on that page is also a Chuck Jones interactive SWF Animation ... sort of a 'One Trick Pony' thing ... )

I'll be glad to host any games you want tested out. 

Note: I have NOT Had a chance to check them ALL Out yet but the Arcade Games all appear to be working MUCH Faster and minor errors are all gone. As I check out the games already there and post new ones to "Play With" I'll be SURE To file prompt bug reports IF and When I find them.

Any Suggestions for games YOU'D Like to Play ... to show up on that page? We have a Chess Program and a Nice Sonic Hedgehog Arcade ... Maukie the Cat is STILL One of my all time favorite SWF Files! It fills 90% of my "Cat" needs.

SWF can be VERY Therapeutic ... the work of the Ruffle Project is one of the most worthy and under sung projects out there. People still NEED their SWFs ... you Devs are literally My Heroes!!!! 

And YES I posted several links to Ruffle on my MAIN Domain to spread the word. (And a "Test" arcade on my NOT-MAIN Domain to Beta Test and have fun with Files that I Had Missed!!!
